File: expense-tracker/backend/src/main/java/com/expensetracker/service/TransactionService.java
package com.expensetracker.service;

import com.expensetracker.dto.request.TransactionRequest;
import com.expensetracker.dto.response.TransactionResponse;

import java.time.LocalDate;
import java.util.List;

public interface TransactionService {
    TransactionResponse create(Long userId, TransactionRequest request);
    List<TransactionResponse> getAll(Long userId, LocalDate startDate, LocalDate endDate, Long categoryId);
    TransactionResponse getById(Long userId, Long id);
    TransactionResponse update(Long userId, Long id, TransactionRequest request);
    void delete(Long userId, Long id);
}

File: expense-tracker/frontend/src/services/transactionService.js
import api from './api';

const transactionService = {
  async getAll(filters = {}) {
    const { data } = await api.get('/transactions', { params: filters });
    return data;
  },

  async getById(id) {
    const { data } = await api.get(`/transactions/${id}`);
    return data;
  },

  async create(payload) {
    const { data } = await api.post('/transactions', payload);
    return data;
  },

  async update(id, payload) {
    const { data } = await api.put(`/transactions/${id}`, payload);
    return data;
  },

  async remove(id) {
    await api.delete(`/transactions/${id}`);
  },
};

export default transactionService;
